File: 序号/2136. 全部开花的最早一天.kt
class Solution {
  fun earliestFullBloom(plantTime: IntArray, growTime: IntArray): Int {
    // 返回全部开花最早的一天
    val n = plantTime.size
    // 先种长得慢的
    val q = PriorityQueue<IntArray>{
      i,j->
      if(i[0]==j[0])  i[1]-j[1]
      else  j[0]-i[0]
    }
    for(i in 0 until n){
      q.add(intArrayOf(growTime[i],i))
    }
    var ans = 0
    var day = 0
    while(q.isNotEmpty()){
      val (time,idx) = q.poll()!!
      day+=plantTime[idx]
      ans= maxOf(ans,day+time)
    }
    return ans
  }
}

File: 序号/2136. 全部开花的最早一天.py
class Solution:
    def earliestFullBloom(self, plantTime: List[int], growTime: List[int]) -> int:
        # 播种只能一个个的种，种完所有种子天数是固定的，想开花时间最早，需要将生长时间长的最先播
        for i,x in enumerate(growTime):
            growTime[i]=(x,i)
        growTime.sort(reverse=True)
        ans=p=0
        for grow,idx in growTime:
            p+=plantTime[idx]
            ans=max(ans,p+grow)
        return ans